network/peer/tls_config_test.go

```diff
@@ -0,0 +1,131 @@
+	"golang.org/x/crypto/ed25519"
+
+	"github.com/ava-labs/avalanchego/network/peer"
+	"github.com/ava-labs/avalanchego/staking"
+)
+
+func TestValidateCertificate(t *testing.T) {
+	for _, testCase := range []struct {
+		description string
+		input       func(t *testing.T) tls.ConnectionState
+		expectedErr error
+	}{
+		{
+			description: "Valid TLS cert",
+			input: func(t *testing.T) tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+				x509Cert := makeRSACertAndKey(t, key)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509Cert.cert}}
+			},
+		},
+		{
+			description: "No TLS certs given",
+			input: func(*testing.T) tls.ConnectionState {
+				return tls.ConnectionState{}
+			},
+			expectedErr: peer.ErrNoCertsSent,
+		},
+		{
+			description: "Empty certificate given by peer",
+			input: func(*testing.T) tls.ConnectionState {
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{nil}}
+			},
+			expectedErr: peer.ErrEmptyCert,
+		},
+		{
+			description: "nil RSA key",
+			input: func(t *testing.T) tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+
+				x509CertWithNilPK := makeRSACertAndKey(t, key)
+				x509CertWithNilPK.cert.PublicKey = (*rsa.PublicKey)(nil)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509CertWithNilPK.cert}}
+			},
+			expectedErr: staking.ErrInvalidRSAPublicKey,
+		},
+		{
+			description: "No public key in the cert given",
+			input: func(t *testing.T) tls.ConnectionState {
+				key, err := rsa.GenerateKey(rand.Reader, 2048)
+				require.NoError(t, err)
+
+				x509CertWithNoPK := makeRSACertAndKey(t, key)
+				x509CertWithNoPK.cert.PublicKey = nil
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{&x509CertWithNoPK.cert}}
+			},
+			expectedErr: peer.ErrEmptyPublicKey,
+		},
+		{
+			description: "EC cert",
+			input: func(t *testing.T) tls.ConnectionState {
+				ecKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+				require.NoError(t, err)
+
+				basicCert := basicCert()
+				certBytes, err := x509.CreateCertificate(rand.Reader, basicCert, basicCert, &ecKey.PublicKey, ecKey)
+				require.NoError(t, err)
+
+				ecCert, err := x509.ParseCertificate(certBytes)
+				require.NoError(t, err)
+				return tls.ConnectionState{PeerCertificates: []*x509.Certificate{ecCert}}
+			},
+		},
+		{
+			description: "EC cert with empty key",
+			expectedErr: peer.ErrEmptyPublicKey,
+			input: func(t *testing.T) tls.ConnectionState {
+				ecKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+				require.NoError(t, err)
+
+				basicCert := basicCert()
+				certBytes, err := x509.CreateCertificate(rand.Reader, basicCert, basicCert, &ecKey.PublicKey, ecKey)
+				require.NoError(t, err)
+
+				ecCert, err := x509.ParseCertificate(certBytes)
+				require.NoError(t, err)
+
+				ecCert.PublicKey = nil
```

Comment:
Is this supposed to hit the ecdsa public key nil check?
```suggestion
				ecCert.PublicKey = (*ecdsa.PublicKey)(nil)
```

Reply:
yes